language/vm/vm_runtime/src/data_cache.rs

```diff
@@ -246,9 +246,23 @@ impl<'txn> TransactionDataCache<'txn> {
                 let deleted = global_ref.is_deleted();
                 if let Some(data) = global_ref.get_data() {
                     if deleted {
-                        write_set.push((key, WriteOp::Deletion));
+                        if write_set.len() < usize::max_value() {
```

Comment:
If write_set.len() == usize::max_value, all bytes in addressable memory are in use in write_set, which is patently absurd.  There is no room left for code, a stack or even the pointer to the write_set itself.

I think using assume! would be a better choice than creating an unreachable error path.

Reply:
Agreed. It bears asking, however, if it is desirable to allow write sets to grow to the point where out of memory errors can occur. In this case I suspect that the gas limit will prevent the write set from getting close. If so, that can be used as the justification for the assumption.

Reply:
Thanks for pointing that out. I've updated it to use an assumption rather than a check.